File: typescript/01/src/instructionParser.ts
export class Instruction {
  public dir: string;
	public count: string;

  constructor(dir: string, count: string) {
    this.dir = dir;
    this.count = count;
  }
}

export function parseInstructions(list: string): Instruction[] {
  return list.split(/[\n,]\s*/).map(entry => {
    dir = entry.split("")[0];
    count = entry.substring(1);
    return new Instruction(dir, count);
  });
}

File: ruby/01/lib/instruction_parser.rb
class InstructionParser
  attr_reader :instructions

  Instruction = Struct.new(:rotation, :clicks)

  def initialize(list)
    @instructions = parse(list)
  end

  private

  def parse(list)
    list.split(/[\n,]\s*/).filter_map do |entry|
      rotation = entry[/\A\s*(?:L|R)/i]
      clicks = entry[/\d+\z/]

      next unless rotation && clicks

      Instruction.new(rotation.strip.upcase, clicks.to_i)
    end
  end
end
